src/devices/sound/c6280.cpp

```diff
@@ -114,18 +114,45 @@ void c6280_device::sound_stream_update(sound_stream &stream, stream_sample_t **i
 			}
 			else
 			{
-				/* Waveform mode */
-				uint32_t step = m_wave_freq_tab[m_channel[ch].m_frequency];
-				for (int i = 0; i < samples; i += 1)
+				if ((m_lfo_control & 0x80) && ((ch == 0) && (ch != 1)))
```

Comment:
The `(ch != 1)` check is redundant - if `ch` is equal to zero, it definitely won't be equal t 1